anomalib/models/components/base/anomaly_module.py

```diff
@@ -58,12 +57,9 @@ def __init__(self, params: Union[DictConfig, ListConfig]):
         self.model: nn.Module
 
         # metrics
-        image_auroc = AUROC(num_classes=1, pos_label=1, compute_on_step=False)
-        image_f1 = F1(num_classes=1, compute_on_step=False, threshold=self.hparams.model.threshold.image_default)
-        pixel_auroc = AUROC(num_classes=1, pos_label=1, compute_on_step=False)
-        pixel_f1 = F1(num_classes=1, compute_on_step=False, threshold=self.hparams.model.threshold.pixel_default)
-        self.image_metrics = MetricCollection([image_auroc, image_f1], prefix="image_").cpu()
-        self.pixel_metrics = MetricCollection([pixel_auroc, pixel_f1], prefix="pixel_").cpu()
+        self.image_metrics, self.pixel_metrics = get_metrics(self.hparams)
```

Comment:
Is there something planned for classification models which do not have pixel metrics? When I removed the pixel metric key from the config file it threw error for padim

Reply:
This should work:
```
metrics:
  image:
    - F1
    - AUROC
  pixel: []
```

Or this:
  ```
  metrics:
  image:
    - F1
    - AUROC
  pixel: None

Reply:
Works now 🙂

Reply:
It would be good to handle a case such as following:
```yaml
metrics:
  image:
    - F1
    - AUROC
```
so `pixel: []` would be assigned under the hood.

Reply:
It would be good to handle a case such as following:
```yaml
metrics:
  image:
    - F1
    - AUROC
```
so `pixel: []` would be assigned under the hood.

Reply:
Or what would be the behaviour for the following:
```yaml
metrics:
 - F1
 - AUROC
```

Reply:
Or what would be the behaviour for the following:
```yaml
metrics:
 - F1
 - AUROC
```

Reply:
Not sure why my comments disappeared from here, but ideally 
```yaml
metrics:
  image:
    - F1
    - AUROC
```

should also work fine.